src/core/memory.d

```diff
@@ -919,3 +919,186 @@ extern (C) private pure @system @nogc nothrow
 
     pragma(mangle, "free") void fakePureFree(void* ptr);
 }
+
+private extern (C) void _d_delinterface(void** p);
+private extern (C) void _d_delclass(Object* p);
+private extern (C) void _d_delstruct(void** p, TypeInfo_Struct inf);
+
+/**
+The `delete` keyword allowed to free GC-allocated memory.
+As this is inherently not `@safe`, it has been deprecated.
+This function has been added to provide an easy transition from `delete`.
+It performs the same functionality as `delete`.
+
+Note: Users should prefer $(REF destroy, object)` to explicitly finalize objects,
+and only resort to $(REF __delete, core,memory) when $(REF destroy, object)
+would not be a feasible option.
+
+In detail, `__delete` does the following:
+$(UL
+
+    $(LI If `t` is a class object reference, and
+        there is a destructor for that class, the destructor
+        is called for that object instance.
+    )
+    $(LI Otherwise, the garbage collector is called to immediately free the
+        memory allocated for the class instance or struct instance.
+        If the garbage collector was not used to allocate the memory for
+        the instance, undefined behavior will result.
+    )
+    $(LI If `t` is a pointer or a dynamic array,
+        the garbage collector is called to immediately release the
+        memory.
+        If the garbage collector was not used to allocate the memory for
+        the instance, undefined behavior will result.
+    )
```

Comment:
The implementation just sets the pointer to null.  I don't think that initiates an immediate call to GC to free the memory.